app/src/main/java/com/github/libretube/api/NewPipeMediaServiceRepository.kt

```diff
@@ -319,14 +321,50 @@ class NewPipeMediaServiceRepository : MediaServiceRepository {
         )
     }
 
+    @OptIn(ExperimentalStdlibApi::class)
     override suspend fun getSegments(
         videoId: String,
-        category: String,
-        actionType: String?
-    ): SegmentData = SegmentData()
+        category: List<String>,
+        actionType: List<String>?
+    ): SegmentData {
+        // use hashed video id for privacy
+        // https://wiki.sponsor.ajay.app/w/API_Docs#GET_/api/skipSegments/:sha256HashPrefix
+        val hashedId = MessageDigest.getInstance("SHA-256")
+            .digest(videoId.toByteArray())
+            .toHexString()
+
+        return RetrofitInstance.externalApi.getSegments(
+            hashedId.substring(0..4),
+            category,
+            actionType
+        ).first { it.videoID == videoId }
+    }
 
-    override suspend fun getDeArrowContent(videoIds: String): Map<String, DeArrowContent> =
-        emptyMap()
+    @OptIn(ExperimentalStdlibApi::class)
+    override suspend fun getDeArrowContent(videoIds: String): Map<String, DeArrowContent> {
+        // use hashed video id for privacy
+        // https://wiki.sponsor.ajay.app/w/API_Docs/DeArrow#GET_/api/branding/:sha256HashPrefix
+        val digest = MessageDigest.getInstance("SHA-256")
+        return videoIds.split(',').chunked(25)
+            .flatMap {
+                it.parallelMap { videoId ->
+                    val hashedId = digest
+                        .digest(videoId.toByteArray())
+                        .toHexString()
+                    RetrofitInstance.externalApi.getDeArrowContent(
```

Comment:
Maybe we should use `runCatching` here because if one chunk of data fails to load (for whatever reasons, e.g. ratelimits), we could still display the results from the other requests.